lib/jxl/decode.cc

```diff
@@ -1743,6 +1746,23 @@ JxlDecoderStatus JxlDecoderProcessInput(JxlDecoder* dec) {
                sizeof(dec->box_decoded_type));
       }
 
+      // Box order validity checks
+      // The signature box at box_count == 1 is not checked here since that's
+      // already done at the beginning.
+      dec->box_count++;
+      if (dec->box_count == 2) {
```

Comment:
can't there be other, non-jxl-related boxes?

Reply:
No, ftyp has to be the first box after the signature box.